play-json/src/main/scala/play/api/libs/json/JsConstraints.scala

```diff
@@ -52,6 +55,19 @@ trait PathReads {
     )
   }
 
+  def optional[A](path: JsPath)(implicit reads: Reads[A]) = Reads[Option[A]] { json =>
+    path.applyTillLast(json).fold(
+      jserr => jserr,
+      jsres => jsres.fold(
+        _ => JsError(path, JsonValidationError("error.path.missing")),
```

Comment:
Please see prev answer. Sending `null` one is still able to reset optional field with default behavour to `None`.